core/gui/src/app/hub/component/workflow/detail/hub-workflow-detail.component.ts

```diff
@@ -174,105 +154,68 @@ export class HubWorkflowDetailComponent implements AfterViewInit, OnDestroy, OnI
-      ? this.workflowPersistService.retrieveWorkflow(wid)
-      : this.hubWorkflowService.retrievePublicWorkflow(wid);
-    workflowObservable.pipe(untilDestroyed(this)).subscribe(
-      (workflow: Workflow) => {
-        this.workflowActionService.setNewSharedModel(wid, this.userService.getCurrentUser());
-        // remember URL fragment
-        const fragment = this.route.snapshot.fragment;
-        // load the fetched workflow
-        this.workflowActionService.reloadWorkflow(workflow);
-        this.workflowActionService.enableWorkflowModification();
-        // set the URL fragment to previous value
-        // because reloadWorkflow will highlight/unhighlight all elements
-        // which will change the URL fragment
-        this.router.navigate([], {
-          relativeTo: this.route,
-          fragment: fragment !== null ? fragment : undefined,
-          preserveFragment: false,
-        });
-        // highlight the operator, comment box, or link in the URL fragment
-        if (fragment) {
-          if (this.workflowActionService.getTexeraGraph().hasElementWithID(fragment)) {
-            this.workflowActionService.highlightElements(false, fragment);
-          } else {
-            this.notificationService.error(`Element ${fragment} doesn't exist`);
-            // remove the fragment from the URL
-            this.router.navigate([], { relativeTo: this.route });
-          }
-        }
-        // clear stack
-        this.undoRedoService.clearUndoStack();
-        this.undoRedoService.clearRedoStack();
-      },
-      () => {
-        this.workflowActionService.resetAsNewWorkflow();
-        // enable workspace for modification
-        this.workflowActionService.enableWorkflowModification();
-        // clear stack
-        this.undoRedoService.clearUndoStack();
-        this.undoRedoService.clearRedoStack();
-        this.message.error("You don't have access to this workflow, please log in with an appropriate account");
-      }
-    );
-  }
-
-  registerLoadOperatorMetadata() {
-    this.operatorMetadataService
-      .getOperatorMetadata()
+    this.hubWorkflowService
+      .retrievePublicWorkflow(wid)
       .pipe(untilDestroyed(this))
-      .subscribe(() => {
-        // load workflow with wid if presented in the URL
-        if (this.wid) {
-          // if wid is present in the url, load it from the backend
-          this.userService
-            .userChanged()
-            .pipe(untilDestroyed(this))
-            .subscribe(() => {
-              this.loadWorkflowWithId(this.wid);
-            });
-        } else {
-          // no workflow to load, pending to create a new workflow
+      .subscribe(
+        (workflow: Workflow) => {
+          // remember URL fragment
+          const fragment = this.route.snapshot.fragment;
+          // load the fetched workflow
+          this.workflowActionService.reloadWorkflow(workflow);
+          this.workflowActionService.disableWorkflowModification();
```

Comment:
redundant? I assume the entire component should never change the modification flag except for the constructor to disable it once for all.